app/services/keycloak/PermsService.scala

```diff
@@ -54,4 +56,38 @@ class PermsService @Inject()(config: Configuration) {
       case e: Throwable => throw e
     }
   }
+
+  def createPermissions(token: String, userName:String, files: Set[String]): (Set[String], Set[String]) = {
+    val protection = authzClient.protection(token)  // insure token has resource creation rights
+    val resources = protection.resource()
+
+    val createdPerms = files.map(file => {
+      val res = resources.findByName(file)
+      if(res != null) {  // resource has to exist
+        if(!res.getOwnerManagedAccess) {  // has to be true
+          res.setOwnerManagedAccess(true)
+          resources.update(res)
+        }
+        val policy = protection.policy(res.getId)
+        val permName = file
+        val existingPerm = policy.find(permName, null,  0 , 1) // should be only one perm with this name
+          .stream().findFirst
+        existingPerm.ifPresentOrElse(perm => {
+          if(!perm.getUsers.contains(userName)){
+            perm.addUser(userName)
+            policy.update(perm)
+          }
+        }, () => {
+          val perm = new UmaPermissionRepresentation
+          perm.setName(permName)
+          perm.addUser(userName)
+          policy.create(perm)
+        })
+        file
+      }else {
+        null
```

Comment:
why not returning an empty set ?

Reply:
this is part of the .map(...) I want file when the creation is OK and null when failed. But I can replace it with flatmap of Option will be cleaner